src/drive.ts

```diff
@@ -232,18 +232,19 @@ async function getFileList(
     '/api/contents/' + path + '' + url,
     { method: 'get' }
   );
-
-  const hiddenFilesNo: number = result.content.filter(
-    (file: { name: string }) => file.name.startsWith('.')
-  ).length;
-  await stateDB.save('hiddenFilesNo', hiddenFilesNo);
-
-  if (!showHidden && Array.isArray(result.content)) {
-    const filteredResult = JSON.parse(JSON.stringify(result));
-    filteredResult.content = (result.content as Array<any>).filter(
-      (file: { name: string }) => !file.name.startsWith('.')
-    );
-    return filteredResult;
+  if (path !== null && !path.includes('.')) {
+    const hiddenFilesNo: number = result.content.filter(
+      (file: { name: string }) => file.name.startsWith('.')
+    ).length;
+    await stateDB.save('hiddenFilesNo', hiddenFilesNo);
+
+    if (!showHidden && Array.isArray(result.content)) {
+      const filteredResult = JSON.parse(JSON.stringify(result));
```

Comment:
Why are we stringify and parse at the same time? If there's a type issue, can't you cast?

Reply:
This is a workaround to clone the object in order to substitute the list of files. Casting to a type won't help, since Contents.IModel is immutable